transformation-scripts/semopenalex-keywords.py

```diff
@@ -0,0 +1,285 @@
+trig_output_file_path = f'{trig_output_dir_path}{ENTITY_TYPE_OUTPUT}-semopenalex-{today}.trig'
+
+data_dump_start_time = time.ctime()
+print('topics entity files started to download at: '+ data_dump_start_time)
+# Copy institutions entity snapshot
+client = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+file_names, folders = get_file_folders(client, "openalex", "data/topics/")
+download_files(client, "openalex", data_dump_input_root_dir, file_names, folders)
+print('topics entity files finished to download.')
+
+start_time = time.ctime()
+print('topics entity started to transform at: '+ start_time)
+
+with open(trig_output_file_path, "w", encoding="utf-8") as g:
+
+    #Path where the OpenAlex data for the current entity type is located
+    data_dump_input_entity_dir = f'{data_dump_input_root_dir}/data/{ENTITY_TYPE_INPUT}/*'
+
+
+    # initialize and add domain scheme URI
+    keywords_graph.add((keyword_scheme_uri, RDF.type, SKOS.ConceptScheme))
+    keywords_graph.add((keyword_scheme_uri, SKOS.prefLabel, Literal("SemOpenAlex Keywords", datatype = XSD.string)))
+    keywords_graph.add((keyword_scheme_uri, URIRef("http://purl.org/dc/terms/description"), Literal("SemOpenAlex keywords are based on SemOpenAlex topics. They are specific words or phrases used to capture the essential topics or ideas of an entity.", datatype = XSD.string)))
+
+
+    for filename in glob.glob(os.path.join(data_dump_input_entity_dir, '*.gz')):
+        with gzip.open(filename, 'r') as f:
+            for line in f:
+                try:
+                    json_data = json.loads(line.decode('utf-8'))
+
+                    # keyword-URI
+                    # keywords
+                    topic_keywords = json_data['keywords']
+                    if not topic_keywords is None:
+                        for keyword in topic_keywords:
+                            replacements_url = clean_url(keyword)
+                            keyword_uri = transform_keyword_to_uri(keyword)
+                            keywords_graph.add((URIRef(keyword_uri), RDF.type, soa_class_keyword))
+                            keywords_graph.add((URIRef(keyword_uri), RDF.type, SKOS.Concept))
+                            keywords_graph.add((URIRef(keyword_uri), SKOS.inScheme, keyword_scheme_uri))
+                            keywords_graph.add((URIRef(keyword_scheme_uri), SKOS.hasTopConcept, keyword_uri))
+                            keywords_graph.add((URIRef(keyword_uri), SKOS.prefLabel, Literal(keyword, datatype=XSD.string)))
+                    
+
+                    i += 1
+                    if i % 100 == 0:
+                        print('Processed topics entity {} lines for the keyword files generation'.format(i))
+
+                    if i % 100 == 0:
+                        g.write(keywords_graph.serialize(format='trig'))
+                        keywords_graph = Graph(identifier=context)
```

Comment:
simply removing this code, it solves the problem for writing several context in trig format. And I guess this is due to a very small file.

Reply:
Done. Because the keywords file will be rather small, the entire data is stored in a rdflib graph during the transformation and only write to the .trig file at the end.